src/Consumption/LimitAttemptsExtension.php

```diff
@@ -58,10 +60,22 @@ public function onResult(MessageResult $context): void
         $attemptNumber = $message->getProperty(self::ATTEMPTS_PROPERTY, 0) + 1;
 
         if ($attemptNumber >= $maxAttempts) {
+            $originalResult = $context->getResult();
+
             $context->changeResult(
                 Result::reject(sprintf('The maximum number of %d allowed attempts was reached.', $maxAttempts))
             );
 
+            $exception = $originalResult instanceof Result ? $originalResult->getReason() : null;
+
+            $event = new Event(
+                'Consumption.LimitAttemptsExtension.failed',
+                $jobMessage,
+                ['exception' => $exception, 'logger' => $context->getLogger()]
+            );
+
+            EventManager::instance()->dispatch($event);
```

Comment:
I think it can actually. The listener can be attached when instantiating the extension in `WorkerCommand`. That's the only place it's instantiated so there isn't really any need for a global event.  I'll make that change.